### ISSUE TITLE
Update per batch, not per epoch 

### ISSUE THREAD

**Issue by GLambard** [titu1994/Snapshot-Ensembles#15]
First of all, thank you very much for sharing your implementation of the Snapshot Ensembling. However, you should not use the `callbacks.LearningRateScheduler` to update the learning rate. 

As mentioned in the [original paper](https://openreview.net/pdf?id=BJYwwY9ll) under the equation 2, "we update the learning rate at each iteration" (i.e batch) "rather than at every epoch" to improve the convergence of short cycles. But, `callbacks.LearningRateScheduler` in Keras, with the [source code](https://github.com/keras-team/keras/blob/master/keras/callbacks.py#L633) here, updates the learning rate at 'on_epoch_begin'. 

After this fix, you should improve further your scores.
